core/capabilities/webapi/trigger.go

```diff
@@ -1,18 +1,150 @@
+	mu        sync.Mutex
+	triggers  map[string]chan capabilities.TriggerResponse
+}
+
+var _ capabilities.TriggerCapability = (*workflowConnectorHandler)(nil)
+var _ services.Service = &workflowConnectorHandler{}
+
 func NewTrigger(config string, registry core.CapabilitiesRegistry, connector connector.GatewayConnector, lggr logger.Logger) (job.ServiceCtx, error) {
 	// TODO (CAPPL-22, CAPPL-24):
 	//   - decode config
 	//   - create an implementation of the capability API and add it to the Registry
 	//   - create a handler and register it with Gateway Connector
 	//   - manage trigger subscriptions
 	//   - process incoming trigger events and related metadata
-	return nil, nil
+
+	handler := &workflowConnectorHandler{
+		connector: connector,
+		lggr:      lggr.Named("WorkflowConnectorHandler"),
+	}
+
+	// is this the right way to register with gateway connector?  Cron trigger doesn't do this.
+	err := connector.AddHandler([]string{"add_workflow"}, handler)
+
+	return handler, err
+}
+
+func (h *workflowConnectorHandler) HandleGatewayMessage(ctx context.Context, gatewayID string, msg *api.Message) {
+	body := &msg.Body
+	fromAddr := ethCommon.HexToAddress(body.Sender)
+	// TODO: apply allowlist and rate-limiting
+	h.lggr.Debugw("handling gateway request", "id", gatewayID, "method", body.Method, "address", fromAddr)
+
+	switch body.Method {
+	case workflow.MethodAddWorkflow:
+		// TODO: add a new workflow spec and return success/failure
+		// we need access to Job ORM or whatever CLO uses to fully launch a new spec
+		h.lggr.Debugw("added workflow spec", "payload", string(body.Payload))
+		response := Response{Success: true}
+		h.sendResponse(ctx, gatewayID, body, response)
+	default:
+		h.lggr.Errorw("unsupported method", "id", gatewayID, "method", body.Method)
+	}
+}
+
+// Register a new trigger
+// Can register triggers before the service is actively scheduling
+func (h *workflowConnectorHandler) RegisterTrigger(ctx context.Context, req capabilities.TriggerRegistrationRequest) (<-chan capabilities.TriggerResponse, error) {
+	// There's no config to use and validate
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	_, ok := h.triggers[req.TriggerID]
+	if ok {
+		return nil, fmt.Errorf("triggerId %s already registered", req.TriggerID)
+	}
+
+	callbackCh := make(chan capabilities.TriggerResponse, defaultSendChannelBufferSize)
+	h.triggers[req.TriggerID] = callbackCh
+
+	return callbackCh, nil
+}
+
+func (h *workflowConnectorHandler) UnregisterTrigger(ctx context.Context, req capabilities.TriggerRegistrationRequest) error {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	trigger := h.triggers[req.TriggerID]
+
+	// Close callback channel
+	close(trigger)
+	// Remove from triggers context
+	delete(h.triggers, req.TriggerID)
+	return nil
+}
+
+func (h *workflowConnectorHandler) Start(ctx context.Context) error {
+	// how does the
+	return h.StartOnce("GatewayConnectorServiceWrapper", func() error {
+		return nil
+	})
+}
+func (h *workflowConnectorHandler) Close() error {
```

Comment:
Since you are using services.StateMachine StartOnce, you should use StopOnce here